presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -481,6 +495,108 @@ public static void multiply(Slice left, Slice right, Slice result)
         pack(result, (int) z0, (int) z1, (int) z2, (int) z3, leftNegative != rightNegative);
     }
 
+    public static void multiply(Slice left, long right, Slice result)
```

Comment:
This looks like a specialization of `multiply(Slice left, Slice right, Slice result)`. Am I right?

If so can we somehow use specialization in the implementation of generic `multiply(Slice left, Slice right, Slice result)`?

Reply:
Can we extract a method from `public static void multiply(Slice left, Slice right, Slice result)` which we could use in implementation of `multiply(Slice left, long right, Slice result)` and `multiply(long left, long right, Slice result)`?

I am bit concerned of amount of complex code that was copied.

Reply:
That can be done with many methods here. The question is how much do we want to do in this PR.
But let's agree on this particular method.
I'm going to rewrite it to a common multiply(long left, long right, Slice result) method and see if this changes performance in any way.

Reply:
> That can be done with many methods here.

Correct, I took this one as an example. If it would work here, then it should work for others.

Reply:
> If it would work here, then it should work for others

With JIT? Dream on.

I made some tests with this approach. Results are interesting - out of 5 benchmarks 3 are the same, one is slightly faster(!) and one is terribly slower(~40%).
I'm going to take care of that in another PR. I'll try to find a common ground between performance and code reusability.